### ISSUE TITLE
Actions where null is provided should not die

### ISSUE THREAD

**Issue by adet4ever** [shipshapecode/ember-flatpickr#415]
In cases where null value is provided to action hooks like onReady, onOpen, onClose, the component initialization just errors out.

for example:
 this.onReady is not a function.

**Comment by rwwagner90** [comment]
@adet4ever those are optional params. Can you share some code that is causing the issue please?

**Comment by adet4ever** [comment]
Hi @rwwagner90 

Here is what I am trying to do. To wrap the component with mine. I created a glimmer component  called date-time-picker
```
import Component from '@glimmer/component';

export default class DateTimePickerComponent extends Component {
}
```
the template like:
```
<EmberFlatpickr
  @altFormat="Y-m-d"
  @altInputClass="my-alt-input"
  @mode="single"
  @dateFormat="Y-m-d"
  @defaultMinute={{0}}
  @defaultHour={{12}}
  @locale="ru"
  @hourIncrement={{1}}
  @minuteIncrement={{5}}
  @placeholder="Choose a Date"
  @date={{@defaultDate}}
  @defaultDate={{@defaultDate}}
  @disable={{@datesToDisable}}
  @enable={{@datesToEnable}}
  @flatpickrRef={{@flatpickrRef}}   
  @maxDate={{@maxDate}}
  @minDate={{@minDate}}
  @getFlatpickrRef={{@getFlatpickrRef}} 
  @onChange={{@onChange}}
  @onOpen={{@onOpen}}
  @onReady={{@onReady}}
  @onClose={{@onClose}}  @enableTime={{true}}
  @disableMobile={{false}}
  @enableSeconds={{false}}
  @clickOpens={{true}}
  @altInput={{true}}
  @inline={{false}}
  @allowInput={{true}}
  @noCalendar={{false}}
  @parseDate={{false}}
  @shorthandCurrentMonth={{false}}
  @static={{false}}
  @time_24hr={{false}}
  @weekNumbers={{false}}
  @prevArrow="<"
  @nextArrow=">"
/>
```
And my usage in another template  as:
 
```
        <DateTimePicker 
          @defaultDate="2020-02-288"
          @date="2020-02-28"
          @onChange={{this.nullItem}}
          @datesToDisable={{ this.datesToDisable }}
          @datesToEnable={{ this.datesToDisable }}
        />
```
This was causing error since the hooks (onReady, onOpen, onClose) are listed/passed in to the components params. Of course it works if I dont list them in the component.

Also according to the config on flatpickr page, those hooks can be null, so when I tried to pass null values, they still break.

I debugged it for a bit to find the problem in the js code in node_modules/flatpickr/dist

```
function triggerEvent(event, data) {
            // If the instance has been destroyed already, all hooks have been removed
            if (self.config === undefined)
                return;
            var hooks = self.config[event];
            if (hooks !== undefined && hooks.length > 0) {
                for (var i = 0; hooks[i] && i < hooks.length; i++)
                    hooks[i](self.selectedDates, self.input.value, self, data);
            }
            if (event === "onChange") {
                self.input.dispatchEvent(createEvent("change"));
                // many front-end frameworks bind to the input event
                self.input.dispatchEvent(createEvent("input"));
            }
        }
```
Since the I listed the hooks, even though with null values, the trigger event function just tries to call the hooks function

`hooks[i](self.selectedDates, self.input.value, self, data);
`
 and in this case does not exists.

It looks like they cannot be listed if a function/action is not passed in.

**Comment by rwwagner90** [comment]
@adet4ever this addon uses a mixin, which is not compatible with glimmer components. We probably need to rethink the implementation.

**Comment by rwwagner90** [comment]
@adet4ever we just released v3.0.0 which is refactored to not use mixins and is now a glimmer component. Please give it a try!
